### PR TITLE
RAC-737: Add Collapse component in DSM

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

image

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+## Playground
+
+<Canvas>
+  <Story name="Standard">
+    {args => {
+      const [{isOpen}, updateArgs] = useArgs();
+      const onCollapse = isOpen => updateArgs({isOpen});
+      return <Collapse {...args} onCollapse={onCollapse} />;
+    }}
```
Comment (juliensnz) [inline front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx:36]: why do we still use "args"?

**Review by juliensnz — APPROVED** [review]

File: front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx (hunk 1 above)
Comment (ValentinMumble) [reply front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx:36]: To update the props controls in the main story directly 🙂

File: front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        <Collapse
+          {...args}
+          isOpen={isOpen}
+          onCollapse={setOpen}
+          label={
+            <>
+              Numerable content <Badge level="secondary">42</Badge>
+            </>
+          }
+        />
```
Comment (StevenVAIDIE) [inline front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx:80]: Something like this is not possible ?
```suggestion
        <Collapse {...args}>
          <Collapse.Title isOpen={isOpen} onCollapse={setOpen}>
                Numerable content
                <Badge level="secondary">42</Badge>
          </Collapse.Title>
          <Collapse.Content isOpen={isOpen}>
             Collapse content
          </Collapse.Content>
     </Collapse>
```

File: front-packages/akeneo-design-system/src/components/Collapse/Collapse.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * The collapse is used to organise groups of elements and possibly hide them.
+ */
+const Collapse = React.forwardRef<HTMLDivElement, CollapseProps>(
+  (
+    {label, collapseButtonLabel, isOpen, onCollapse, children, ...rest}: CollapseProps,
+    forwardedRef: Ref<HTMLDivElement>
+  ) => {
+    const collapse = () => onCollapse(!isOpen);
```
Comment (StevenVAIDIE) [inline front-packages/akeneo-design-system/src/components/Collapse/Collapse.tsx:68]:
```suggestion
    const handleCollapse = () => onCollapse(!isOpen);
```

File: front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx (hunk 2 above)
Comment (ValentinMumble) [reply front-packages/akeneo-design-system/src/components/Collapse/Collapse.stories.mdx:80]: For me, this syntax makes sense when the subcomponents are either optional or can be placed in different orders inside the component among its children, here the label is mandatory, has a fixed location and is not part of the content so I would use it as a prop... Let me know what you think

**Comment by juliensnz** [comment]
I'm not sure the animation is the good one. We should not scale the collapse but we should expand it from the top

**Review by StevenVAIDIE — APPROVED** [review]

**Review by StevenVAIDIE — APPROVED** [review]

**Pull request merged**